scripts/rv_expression_association/get_rv_outliers.py

```diff
@@ -0,0 +1,197 @@
+    logging.info(
+        f'Number of non-ref variants for this indvidual: {donor_mt.count()[0]}'
+    )
+
+    # focus on SNVs for now
+    donor_mt = donor_mt.filter_rows(donor_mt.vep.variant_class == 'SNV')
+    # filter for biallelic only
+    donor_mt = donor_mt.filter_rows(hl.len(donor_mt.alleles) == 2)  # not needed?
+    logging.info(
+        f'Number of variants after filtering for biallelic SNVs: {donor_mt.count()[0]}'
+    )
+
+    # SKIP FOR NOW NO VEP ANNOTATIONS
+    # # filter to only variants with some regulatory consequences
+    # donor_mt = donor_mt.filter_rows(
+    #     hl.len(donor_mt.vep.regulatory_feature_consequences) > 0
+    # )
+    # logging.info(
+    #     f'Number of variants after filtering for variants with regulatory consequences: {donor_mt.count()[0]}'
+    # )
+
+    # retain info about het variants
+    is_het_list = donor_mt.variant_qc.n_het.collect()
+
+    # annotate variants with CADD scores, gnomad etc
+    logging.info('Annotate variants with CADD scores and gnomad AF')
+    # ref_ht = hl.read_table(
+    #     'gs://cpg-reference/seqr/v0-1/combined_reference_data_grch38-2.0.4.ht'
+    # )
+    ref_ht = reference_path('seqr/v0-1/combined_reference_data_grch38-2.0.4.ht')
+    donor_mt = donor_mt.annotate_rows(
+        cadd=ref_ht[donor_mt.row_key].cadd,
+        gnomad_genomes=ref_ht[donor_mt.row_key].gnomad_genomes,
+    )
+
+    # get CADD scores
+    cadd_list = donor_mt.cadd.PHRED.collect()
+
+    # get gnomad MAF (popmax now, need NFE too?)
+    maf_popmax = donor_mt.gnomad_genomes.AF_POPMAX_OR_GLOBAL.collect()
+
+    logging.info('Get AF within OneK1K sample')
+    # get MAF within sample
+    # by selecting the appropriate rows in the full object
+    relevant_loci = donor_mt.row_key.collect()
+    mt = mt.filter_rows(hl.set(relevant_loci).contains(mt.row_key))
+    # calculating their variant QC
+    mt = hl.variant_qc(mt)
+    maf_sample_list = mt.variant_qc.AF[1].collect()
+
+    # # get relevant regulatory info from VEP (biotype)
+    # regulatory_consequences = donor_mt.vep.regulatory_feature_consequences[
+    #     'biotype'
+    # ].collect()
+
+    # get gene info
+    gene_start = gene_df[gene_df['gene_name'] == gene_name]['start']
+    gene_end = gene_df[gene_df['gene_name'] == gene_name]['end']
+    ensembl_gene_id = gene_df[gene_df['gene_name'] == gene_name].index[0]
+
+    logging.info('Preparing results data')
+    results_data = {
+        'onek1k_id': onek1k_id,
+        'cpg_id': cpg_id,
+        'gene_name': gene_name,
+        'variant_id': donor_mt.row_key[0].collect(),
+        'is_het': is_het_list,
+        'cadd': cadd_list,
+        'maf_onek1k': maf_sample_list,
+        'maf_gnomad_popmax': maf_popmax,
+        # 'regulatory_consequences': regulatory_consequences,
+        'ensembl_gene_id': ensembl_gene_id,
+        'chrom': chrom,
+        'gene_start': int(gene_start),
+        'gene_end': int(gene_end),
+    }
+    df = pd.DataFrame(results_data)
```

Comment:
Why not write an annotated Hail table instead? Collecting all data on a single node doesn't scale well -- also CSV is not a great format for a number of reasons (lack of a standard, highly inefficient serialization / deserialization as it's not binary-based, loss of numeric precision, etc.).

Reply:
Sounds fine -- in general, please treat all review comments as suggestions! Certainly not trying to push you to tools that aren't fit for the use case.

FWIW there's also some Hail / R interoperability libraries. I haven't looked at them in detail, but I think you should be able to work with a Hail table directly in R this way, without converting to PLINK first.